File: goormlevel/175909/1. 카드 모으기/카드 모으기.js
// Run by Node.js
const readline = require('readline');

(async () => {
	let rl = readline.createInterface({ input: process.stdin });
	let n, m;
	let set = new Set();
	let cnt = 0;
	let res = -1;
	for await (const line of rl) {
		if (!line) rl.close();
		if (!cnt) {
			const nums = line.split(' ').map(Number);
			n = nums[0];
			m = nums[1];
			++cnt;
		} else {
			set.add(line);
			if (set.size === n) res = cnt;
			else ++cnt;
		}
	}
	console.log(res);
	process.exit();
})();

File: goormlevel/175909/1. 카드 모으기/카드 모으기.py
from collections import Counter

N, M = map(int, input().split())

cards = set()
cnt = 0
res = -1
for i in range(M):
	cards.add(input())
	if len(cards) <= N and res == -1:
		cnt += 1
		if len(cards) == N:
			res = cnt
	
print(res)